### ISSUE TITLE
atan(y, x) returns an ArbFloat instead of an ArbReal

### ISSUE THREAD

**Issue by Joel-Dahne** [JeffreySarnoff/ArbNumerics.jl#20]
`atan(y, x)` is in principle the same as `atan(y/x)` but only the latter returns an ArbReal and the former returns an ArbFloat.
``` julia
julia> x = ArbReal(1)
1.0

julia> y = ArbReal(2)
2.0

julia> z1 = atan(y, x)
1.107148717794090503017065460179

julia> z2 = atan(y/x)
1.107148717794090503017065460179

julia> typeof(z1)
ArbFloat{128}

julia> typeof(z2)
ArbReal{128}
```

The fix should be to just remove the midpoint-call in
``` julia
function atan(y::ArbReal{P}, x::ArbReal{P}, prec::Int=P) where {P}
    z = ArbReal{P}()
    ccall(@libarb(arb_atan2), Cvoid, (Ref{ArbReal}, Ref{ArbReal}, Ref{ArbReal}, Clong), z, y, x, prec)
    return midpoint_byref(z)
end
```
and add it to 
``` julia
@inline function atan(y::ArbFloat{P}, x::ArbFloat{P}, prec::Int=P) where {P}
    xr = ArbReal{P}(x)
    yr = ArbReal{P}(y)
    return atan(yr, xr, prec)
end
```

**Comment by JeffreySarnoff** [comment]
Good catch. Thank you.  I will repair this tonight.

**Comment by JeffreySarnoff** [comment]
The delay was unintentional -- JuliaCon distracted me.  This is fixed in master and in v0.6.4 (merge pending).
